tools/str_tools_test.go

```diff
@@ -71,6 +71,35 @@ func TestQuotedFields(t *testing.T) {
 	}
 }
 
+type ShellQuoteTestCase struct {
+	Given    []string
+	Expected []string
+}
+
+func (c *ShellQuoteTestCase) Assert(t *testing.T) {
+	actual := ShellQuote(c.Given)
+
+	assert.Equal(t, c.Expected, actual,
+		"tools: expected ShellQuote(%q) to equal %#v (was %#v)",
+		c.Given, c.Expected, actual,
+	)
+}
+
+func TestShellQuote(t *testing.T) {
+	for desc, c := range map[string]ShellQuoteTestCase{
+		"simple":         {[]string{"foo", "bar", "an_id"}, []string{"foo", "bar", "an_id"}},
+		"leading space":  {[]string{" foo", "bar"}, []string{"' foo'", "bar"}},
+		"trailing space": {[]string{"foo", "bar "}, []string{"foo", "'bar '"}},
+		"internal space": {[]string{"foo bar", "baz quux"}, []string{"'foo bar'", "'baz quux'"}},
+		"backslash":      {[]string{`foo\bar`, `b\az`}, []string{`'foo\bar'`, `'b\az'`}},
+		"quotes":         {[]string{`foo"bar`, "b'az"}, []string{`'foo"bar'`, "'b'\\''az'"}},
+		"mixed quotes":   {[]string{`"foo'ba\"r\"'"`}, []string{`'"foo'\''ba\"r\"'\''"'`}},
+	} {
+		t.Log(desc)
```

Comment:
Nit; this line can be removed, since `go test -v` will log the name of the test for you, like so:

```ShellSession
$ make PKGS=tools GO_TEST_EXTRA_ARGS='-v -run TestShellQuote' test
go test -v -run TestShellQuote ./tools
=== RUN   TestShellQuote
=== RUN   TestShellQuote/quotes
=== RUN   TestShellQuote/mixed_quotes
=== RUN   TestShellQuote/simple
=== RUN   TestShellQuote/leading_space
=== RUN   TestShellQuote/trailing_space
=== RUN   TestShellQuote/internal_space
=== RUN   TestShellQuote/backslash
--- PASS: TestShellQuote (0.00s)
    str_tools_test.go:98: quotes
    --- PASS: TestShellQuote/quotes (0.00s)
    str_tools_test.go:98: mixed quotes
    --- PASS: TestShellQuote/mixed_quotes (0.00s)
    str_tools_test.go:98: simple
    --- PASS: TestShellQuote/simple (0.00s)
    str_tools_test.go:98: leading space
    --- PASS: TestShellQuote/leading_space (0.00s)
    str_tools_test.go:98: trailing space
    --- PASS: TestShellQuote/trailing_space (0.00s)
    str_tools_test.go:98: internal space
    --- PASS: TestShellQuote/internal_space (0.00s)
    str_tools_test.go:98: backslash
    --- PASS: TestShellQuote/backslash (0.00s)
PASS
ok      github.com/git-lfs/git-lfs/tools        0.022s
```

Reply:
I think that I should have added some more detail here. This line:

> ```
>     str_tools_test.go:98: quotes
> ```

is caused by the call to `t.Log()`, whereas this line:

> ```
>     --- PASS: TestShellQuote/quotes (0.00s)
> ```

is caused by the call to `t.Run()`, hence removing the call to `t.Log()` (and the line printed because of it) should still leave enough information to an interested caller of `make test` to figure out what's going on.

Reply:
Sure, I'll do that.